pkg/controller/ps/telemetry.go

```diff
@@ -0,0 +1,123 @@
+package ps
+
+import (
+	"context"
+	"fmt"
+
+	apiv1alpha1 "github.com/percona/percona-server-mysql-operator/api/v1alpha1"
+	"github.com/percona/percona-server-mysql-operator/pkg/telemetry"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func (r *PerconaServerMySQLReconciler) reconcileScheduledTelemetrySending(ctx context.Context, cr *apiv1alpha1.PerconaServerMySQL) error {
+	logger := logf.FromContext(ctx).WithName("reconcileScheduledTelemetrySending")
+
+	jn := telemetryJobName(cr)
+	existingJob, ok := r.Crons.telemetryJobs.Load(jn)
+	if !telemetryEnabled() {
+		if ok {
+			r.Crons.telemetryJobs.Delete(jn)
+		}
+		return nil
+	}
+	job := telemetryJob{}
+	if ok {
+		job = existingJob.(telemetryJob)
+	}
+
+	configuredSchedule := telemetry.Schedule()
+
+	// If the configured schedule is identical to the existing job's schedule, no action is needed.
+	if job.cronSchedule == configuredSchedule {
+		return nil
+	}
+
+	logger.Info("removing existing telemetry job because the configured schedule changed", "old", job.cronSchedule, "new", configuredSchedule)
+	r.Crons.telemetryJobs.Delete(jn)
+
+	id, err := r.Crons.addFuncWithSeconds(configuredSchedule, r.telemetrySendingHandlerFunc(ctx, cr, jn))
+	if err != nil {
+		return err
+	}
+
+	logger.Info("adding new job", "name", jn, "schedule", configuredSchedule)
+
+	r.Crons.telemetryJobs.Store(jn, telemetryJob{
+		scheduleJob:  scheduleJob{jobID: id},
+		cronSchedule: configuredSchedule,
+	})
+
+	if cr.Status.State != apiv1alpha1.StateReady {
+		logger.Info("cluster is not ready yet")
+		return nil
+	}
+
+	logger.Info("sending telemetry on cluster start")
+
+	err = r.sendTelemetry(ctx, cr)
+	if err != nil {
+		logger.Error(err, "failed to send telemetry report")
+	}
```

Comment:
does this ever send telemetry? from the logic above it seems no to me

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/1033/commits/213a0411eb4589984643fbcd7aa35a890b9b9420